buyer_data/utils/loan_math.py

```
@@ -0,0 +1,51 @@
+    if required and value is None: raise ValueError(f'Missing required query parameter: {key}')
+    try: return type_cast(value)
+    except (TypeError, ValueError): raise ValueError(f"Invalid value for '{key}': Expected {type_cast.__name__}")
+
+def calculate_interest_power(rate, tenure_months):
+    if tenure_months <= 0: raise ValueError(f'Invalid tenure months: {tenure_months}')
+    if rate == 0: return 1
+    return pow(1 + rate, tenure_months)
+
+def calculate_emi(principal, rate, tenure_months):
+    if principal < 0: raise ValueError('Principle must be positive.')
+    if tenure_months <= 0: return ValueError('Tenure months must be positive')
```

Ruff (0.12.2)

Multiple statements on one line (colon)

(E701)